packages/app/src/cli/models/app/loader.ts

```diff
@@ -639,18 +598,12 @@ class AppLoader<TConfig extends AppConfiguration, TModuleSpec extends ExtensionS
         return !configKeysThatAreNeverModules.includes(key)
       })
 
-    // make some extension instances for the unused keys
-    const unusedExtensionInstances = unusedKeys.map((key) => {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const specConfiguration = {[key]: (appConfiguration as any)[key]}
-      return this.createExtensionInstance(key, specConfiguration, appConfiguration.path, directory)
-    })
-
-    // return all the non null extension instances, plus the unused ones
-    const nonNullExtensionInstances: ExtensionInstance[] = extensionInstancesWithKeys
+    if (unusedKeys.length > 0) {
+      throw new Error('Not all of the config was consumed')
```

Comment:
This error could be confusing for partners no?
Do we need to use all keys in the toml? should we show a better message if we aren't?